### PR TITLE
Add a parameter to control the amount of time to wait for a docker command

### PR DESCRIPTION
**Pull request by turkeylurkey**

### CODE REVIEW

File: docs/libertyDev.md [hunk 1]
```diff
@@ -141,4 +141,5 @@ These can also be specified as command line parameters in addition to the ones i
...
-| dockerfile | File | 3.1-M1 (tech preview) | Location of a Dockerfile to be used by dev mode to build the container that runs your server and to specify the context used to build the container image. The default location is `Dockerfile` in the project root. Setting this parameter overrides the `container` parameter to `true`. | No |
\ No newline at end of file
+| dockerfile | File | 3.1-M1 (tech preview) | Location of a Dockerfile to be used by dev mode to build the container that runs your server and to specify the context used to build the container image. The default location is `Dockerfile` in the project root. Setting this parameter overrides the `container` parameter to `true`. | No |
+| dockerTimeout | integer | 3.1-M2 (tech preview) | Specifies the amount of time dev mode shall wait for the completion of docker operations such as to build the container. The value is specified in seconds and may be between 1 and 3600 inclusive. The default value is 60 seconds. | No |
```
Comment (ericglau) [inline docs/libertyDev.md:145]: See https://github.com/OpenLiberty/ci.maven/pull/907#discussion_r472253516

File: src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy [hunk 2]
```diff
@@ -924,11 +943,16 @@ class DevTask extends AbstractServerTask {
...
+
             if (dockerRunOpts != null) {
                 setContainer(true);
                 return;
             }
+
+            if (dockerTimeout != null) {
+                setContainer(true);
+                return;
+            }
```
Comment (ericglau) [inline src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy:955]: If https://github.com/OpenLiberty/ci.maven/pull/907#discussion_r472243409 will be changed to not set the container option, then remove this part

File: docs/libertyDev.md (hunk 1 above)
Comment (turkeylurkey) [reply docs/libertyDev.md:145]: Agreed

File: src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy (hunk 2 above)
Comment (turkeylurkey) [reply src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy:955]: Agreed

File: docs/libertyDev.md [hunk 3]
```diff
@@ -142,4 +142,4 @@ These can also be specified as command line parameters in addition to the ones i
...
 | dockerfile | File | 3.1-M1 (tech preview) | Location of a Dockerfile to be used by dev mode to build the container that runs your server and to specify the context used to build the container image. The default location is `Dockerfile` in the project root. Setting this parameter overrides the `container` parameter to `true`. | No |
-| dockerTimeout | integer | 3.1-M2 (tech preview) | Specifies the amount of time dev mode shall wait for the completion of docker operations such as to build the container. The value is specified in seconds and may be between 1 and 3600 inclusive. The default value is 60 seconds. | No |
\ No newline at end of file
+| dockerTimeout | integer | 3.1-M2 (tech preview) | Maximum time to wait (in seconds) for the completion of docker operations such as to build the container. The value must be an integer greater than 0. The default value is `60` seconds. | No |
```
Comment (ericglau) [inline docs/libertyDev.md:145]: See https://github.com/OpenLiberty/ci.maven/pull/907#discussion_r473481723

Comment (turkeylurkey) [reply docs/libertyDev.md]: Changed.

File: docs/libertyDev.md [hunk 4]
```diff
@@ -141,4 +141,5 @@ These can also be specified as command line parameters in addition to the ones i
...
-| dockerfile | File | 3.1-M1 (tech preview) | Location of a Dockerfile to be used by dev mode to build the container that runs your server and to specify the context used to build the container image. The default location is `Dockerfile` in the project root. Setting this parameter overrides the `container` parameter to `true`. | No |
\ No newline at end of file
+| dockerfile | File | 3.1-M1 (tech preview) | Location of a Dockerfile to be used by dev mode to build the container that runs your server and to specify the context used to build the container image. The default location is `Dockerfile` in the project root. Setting this parameter overrides the `container` parameter to `true`. | No |
+| dockerBuildTimeout | integer | 3.1-M2 (tech preview) | Maximum time to wait (in seconds) for the completion of the docker operation to build the container. The value must be an integer greater than 0. The default value is `60` seconds. | No |
```
Comment (TrevCraw) [inline docs/libertyDev.md:145]: Same as https://github.com/OpenLiberty/ci.maven/pull/907#discussion_r474249405

Comment (turkeylurkey) [reply docs/libertyDev.md]: Fixed in both places.

**Pull request merged**